models/evc.py

```diff
@@ -1226,26 +1226,27 @@ def check_list_traces(list_circuits):
         circuit_by_traces = {}
         circuits_checked = {}
 
-        for trace_switch in traces:
-            for trace in traces[trace_switch]:
-                id_trace = str(trace[0]['dpid']) + ':' + str(trace[0]['port'])
-                if 'vlan' in trace[0]:
-                    id_trace += ':' + str(trace[0]['vlan'])
-                circuit_from_data = circuit_data.get(id_trace)
-                if circuit_from_data is None:
-                    continue
-                circuit_id = circuit_from_data['circuit_id']
-                trace_name = circuit_from_data['trace_name']
-                circuit = list_circuits[circuit_id]
-                if circuit_id not in circuit_by_traces:
-                    circuit_by_traces[circuit_id] = {}
-                circuit_by_traces[circuit_id][trace_name] = trace
-
-                if 'trace_a' in circuit_by_traces[circuit_id] \
-                        and 'trace_z' in circuit_by_traces[circuit_id]:
-                    circuits_checked[circuit_id] = EVCDeploy.check_trace(
-                        circuit, circuit_by_traces
-                    )
+        for trace in traces["result"]:
```

Comment:
@gretelliz, that's a great start, we can get rid of a bunch of extra data structure transformation that historically used to be here though, let me explain:

With your changes on bulk `PUT /traces` we have a guarantee that the responses will be ordered, so returned response from `EVCDeploy.run_bulk_sdntraces(uni_list)` if you iterate for each 2 subsequent values `for i, j in zip(traces["result"], traces["result"][1:])` then you already have the trace response for the uni_a and uni_z for each circuit. Finally, you'd have to adapt `EVCDeploy.check_trace(circuit, circuit_by_traces)` to both traces instead of take an entire dict to take `EVCDeploy.check_trace(circuit, trace_a, trace_z)`, removing entirely the dictionary dependency, that way you as you're iterating on each uni pair you can start building your `circuit_checked` response. As you adapt `EVCDeploy.check_trace`, notice that it's always going to expect to validate the entire pair, so it's important to pass along a failed trace `[]` empty list as well and handle it (and then in the future on version `2023.1` you can replace the empty list with the expected `incomplete` state that you're implementing).

You can nuke all of the `circuit_data` and `circuit_by_traces` dictionary helpers https://github.com/kytos-ng/mef_eline/blob/fix_branch/models/evc.py#L1203-L1251, we don't need these transformation helpers anymore. Do you see what I mean? It'll be way less code for us to maintain and reason about.

Reply:
@viniarck, you're right. I just uploaded a new version with the changes you suggested.